### DISCUSSION TITLE
Can a MACRO insert something into the active file directly?

### DISCUSSION THREAD

**Discussion by holroy** [chhoumann/quickadd#362] [Q&A]
I'm contemplating on using MACRO's to execute various scripts, possibly DataviewJS scripts, and I was wondering whether it's possible to get the macro to insert stuff into the active file, or possibly better into a dedicated log file?

Is this easily achievable somehow, and if so how? I thought I read somewhere that returning stuff would insert into the active file, but it doesn't seem to work like that. Tried with the following function, which does produce the output in the console, but nothing in the active file:

```
async function helloWorld(params) {
  console.log('hello world')
  return "Hello world"
}
```

**Comment by chhoumann** [comment]
The return value of an inline script will be inserted at the location of the script, e.g. in captures / templates.

A macro can imitate this behavior by 1st running a script and then assigning the desired value to a variable, which can 2ndly be used in a capture.

**Reply by holroy** [reply]
Would it be possible to ask you to write a simple example on this, as I'm not currently seeing how to do this. I kind of understand what you're saying, but I'm having a bit of problem translating it into actual stuff in the Quick Add setup.

**Reply by chhoumann** [reply]
No problem, @holroy.

```js
module.exports = async (QuickAdd) => {
    // Run the logic of the script, e.g.:
    const result = await dataviewApi.doSomething();

    // At the end, we set the variables. These can be used in the rest of the macro.
    // E.g. in captures, in other scripts, or templates.
    QuickAdd.variables = {
        // If you've used variables in prior steps in your macros, it's good to write:
        ...QuickAdd.variables,

        // But here's the one we're setting:
        myVariable: result, // Now ``{{VALUE:myVariable}}`` will be replaced with the result of the script.
    }
};
```

This represents a basic script to do some logic, and then using the result of that logic to assign a value to a variable.
These variables can then be used in subsequent steps in a macro, i.e. any capture/template/script used after will have access to that variable & it's value.
